### PR TITLE
Add clientUpdateByPrimaryKeyMethodGenerated method

### PR DESCRIPTION
**Pull request by octoape**
The motivation for adding this method can be found in https://github.com/mybatis/generator/issues/1116

### CODE REVIEW

File: core/mybatis-generator-core/src/main/java/org/mybatis/generator/api/Plugin.java [hunk 1]
```diff
@@ -2044,4 +2044,30 @@ default boolean clientUpdateByPrimaryKeyMethodGenerated(KotlinFunction kotlinFun
     default boolean shouldGenerate(IntrospectedTable introspectedTable) {
         return true;
     }
+
+    /**
+     * The motivation for adding this method can be found in
+     * https://github.com/mybatis/generator/issues/1116
+     *
+     * This method is called when the updateByPrimaryKey method
+     * has been generated in the client interface.
```
Comment (jeffgbutler) [inline core/mybatis-generator-core/src/main/java/org/mybatis/generator/api/Plugin.java:2053]: Please add a comment stating this method is only called in the dynamic SQL runtime.

**Review by jeffgbutler — COMMENTED** [review]
1. ~~Please make the corresponding change in the Kotlin generator~~ disregard, the Kotlin method is already named correctly
2. You also need to implement new plugin methods in https://github.com/mybatis/generator/blob/master/core/mybatis-generator-core/src/main/java/org/mybatis/generator/api/CompositePlugin.java

**Comment by octoape** [comment]
OK, the code has been changed.

**Comment by jeffgbutler** [comment]
Thanks for your contribution!

**Pull request merged**